### ISSUE TITLE
Tor False Positives from SSL Dissector (bigram check)

### ISSUE THREAD

**Issue by dsokoloski** [ntop/nDPI#578]
The following Tor bigram check from the SSL dissector results in normal HTTPS flows being detected as NDPI_PROTOCOL_TOR:

https://github.com/ntop/nDPI/blob/d9c963061bbe561af4ea1e223fc4589fe333570d/src/lib/protocols/ssl.c#L390

A few examples of domains that would be incorrectly classified as Tor:

- fo**xb**usiness.com
- tri**bd**ss.com
- ta**mg**rt.com
- **jc**rew.com

We disable this call in our fork.  I'm not sure what if anything should be done officially, but I thought opening this issue would be helpful for anyone else trying to understand why some "normal" SSL flows are being detected incorrectly as Tor.

I personally think the bigram check is useless in this context.

**Comment by kYroL01** [comment]
IMHO you're right on this behaviour

**Comment by lucaderi** [comment]
Please provide a pull request

**Comment by dsokoloski** [comment]
@lucaderi That does what?  Disables Tor bigram checks from the SSL dissector?

**Comment by dsokoloski** [comment]
@lucaderi Thanks for excluding those bigrams, but unfortunately they were only examples of domains that match the "impossible" bigram check.

Here is a [link to a summary spreadsheet](https://docs.google.com/spreadsheets/d/1neBG6UF__tcAP5-ZgsXGKNflKTgx5rv3ndxNrCctlN8/edit?usp=sharing) that shows the number of domains from our database of 2,735,863 rows that match the remaining "impossible" bigrams.

Basically, every "impossible" bigram exists many, many times in Internet domains.  Therefore, this "impossible" bigram check for Tor from the SSL dissector _will always produce false positives_.

I'm not personally bothered by this (I've disabled that check), but I disagree that this issue has been fixed.
